metadata-io/src/main/java/com/linkedin/metadata/search/utils/ESAccessControlUtil.java

```diff
@@ -0,0 +1,266 @@
+        response = Optional.of(builder);
+      }
+    }
+
+    // MATCH_ALL filter present or system user or disabled
+    return response;
+  }
+
+  /**
+   * Given an OperationContext and SearchResult, mark the restricted entities. Currently, the entire
+   * entity is marked as restricted using the key aspect name
+   *
+   * @param searchResult restricted search result
+   */
+  public static void restrictSearchResult(
+      @Nonnull OperationContext opContext, @Nonnull SearchResult searchResult) {
+    restrictSearchResult(opContext, searchResult.getEntities());
+  }
+
+  public static <T extends Collection<SearchEntity>> T restrictSearchResult(
+      @Nonnull OperationContext opContext, T searchEntities) {
+    if (opContext.getOperationContextConfig().getSearchAuthorizationConfiguration().isEnabled()
+        && opContext.getSearchContext().isRestrictedSearch()) {
+      final EntityRegistry entityRegistry = opContext.getEntityRegistry();
+      final String actorUrnStr =
+          opContext.getSessionActorContext().getAuthentication().getActor().toUrnStr();
+      final DisjunctivePrivilegeGroup orGroup =
+          new DisjunctivePrivilegeGroup(
+              ImmutableList.of(new ConjunctivePrivilegeGroup(FILTER_PRIVILEGES)));
+
+      for (SearchEntity searchEntity : searchEntities) {
+        final String entityType = searchEntity.getEntity().getEntityType();
+        final Optional<EntitySpec> resourceSpec =
+            Optional.of(new EntitySpec(entityType, searchEntity.getEntity().toString()));
+        if (!AuthUtil.isAuthorized(
+            opContext.getAuthorizerContext().getAuthorizer(), actorUrnStr, resourceSpec, orGroup)) {
+          final String keyAspectName =
+              entityRegistry.getEntitySpecs().get(entityType.toLowerCase()).getKeyAspectName();
+          searchEntity.setRestrictedAspects(new StringArray(List.of(keyAspectName)));
+        }
+      }
+    }
+    return searchEntities;
+  }
+
+  private static final Function<DataHubPolicyInfo, Boolean> activeMetadataViewEntityPolicyFilter =
+      policy ->
+          policy.getPrivileges() != null
+              && PoliciesConfig.ACTIVE_POLICY_STATE.equals(policy.getState())
+              && PoliciesConfig.METADATA_POLICY_TYPE.equals(policy.getType())
+              && FILTER_PRIVILEGES.stream().anyMatch(priv -> policy.getPrivileges().contains(priv));
+
+  private static Stream<QueryBuilder> streamViewQueries(OperationContext opContext) {
+    return opContext.getSessionActorContext().getPolicyInfoSet().stream()
+        .filter(activeMetadataViewEntityPolicyFilter::apply)
+        .map(
+            policy -> {
+              // Build actor query
+              QueryBuilder actorQuery = buildActorQuery(opContext, policy);
+
+              if (!policy.hasResources()) {
+                // no resource restrictions
+                return actorQuery;
+              } else {
+
+                // No filters or criteria, deny access
```

Comment:
I'm not sure that "deny access" is really correct here. Really more that this policy is just not applying anything, but yeah if no policies apply it would deny access.